pkg/bot/slack.go

```diff
@@ -140,18 +136,49 @@ func (b *Slack) Start(ctx context.Context) error {
 					switch ev := innerEvent.Data.(type) {
 					case *slackevents.AppMentionEvent:
 						b.log.Debugf("Got app mention %s", utils.StructDumper().Sdump(innerEvent))
-						sm := slackMessage{
-							log:             b.log,
-							executorFactory: b.executorFactory,
-							Event:           ev,
-							BotID:           b.botID,
-							client:          b.client,
+						msg := slackMessage{
+							Text:            ev.Text,
+							Channel:         ev.Channel,
+							ThreadTimeStamp: ev.ThreadTimeStamp,
 						}
-						if err := sm.HandleMessage(b); err != nil {
+						if err := b.handleMessage(msg); err != nil {
 							b.log.Errorf("Message handling error: %w", err)
 						}
 					}
 				}
+			case socketmode.EventTypeInteractive:
+				callback, ok := event.Data.(slack.InteractionCallback)
+				if !ok {
+					b.log.Errorf("Invalid event %+v\n", event.Data)
+					continue
+				}
+
+				websocketClient.Ack(*event.Request)
+
+				switch callback.Type {
+				case slack.InteractionTypeBlockActions:
+					b.log.Debugf("Got block action %s", utils.StructDumper().Sdump(callback.ActionCallback.BlockActions))
+
+					if len(callback.ActionCallback.BlockActions) != 1 {
+						fmt.Println("ignoring as too many actions selected")
```

Comment:
Use `b.log` and also change the message - `Ignoring as the number of actions is different from 1` or something like this. I know there shouldn't be the case that the len is 0, but...